packages/web/src/components/search/addons/ImageDropdown.js

```diff
@@ -0,0 +1,297 @@
+import { bool, func, string } from 'prop-types';
+import React, { useEffect, useRef, useState } from 'react';
+import { Container, DropPlaceholder, ErrorMessage, FileInput, Label, ORDivider, PlaceholderText, Preview, PreviewImg, URLInput, StyledDeleteIcon } from './ImageDropdownStyles';
+import { Placeholder } from '../../shared/Icons';
+
+
+const ImageError = () => (
+	<ErrorMessage>
+		<ErrorMessage.Icon />
+		<ErrorMessage.Text>
+			Use an image in one of the formats
+			<code>.jpeg</code>, <code>.png</code>,
+			<code>.webp</code>, <code>.bmp</code> and <code>.tif</code>
+		</ErrorMessage.Text>
+	</ErrorMessage>);
+
+
+/*
+*/
+/**
+ * *
+ * image value is base64 encoded string
+ * onChange is called when a new image is selected or removed
+ * onChange is called when a url is changed or removed
+ *
+ * Image modal is responsible for showing the image preview
+ * Image modal is responsible for taking user input and providing a base64 encoded string
+ *
+ * There are three ways to input an image
+ * 1. OS file browser
+ * 2. Drag and drop
+ * 3. URL
+ *
+ * When an image is selected, the image is converted to a base64 encoded string
+ *
+ * 1. Using the OS file browser
+ *
+ * The file browser maintains an internal state of the file
+ * which is accesible by input.files, where input is the file input element.
+ * When the file input element is unmounted this internal state is lost.
+ * Hence we should always use the imageValue provided by the user for the preview image
+ * On delete, we should also call onChange provided by the user with an empty string.
+ *
+ * 2. Drag and drop
+ *
+ * 3. URL
+ * */
+// eslint-disable-next-line import/prefer-default-export
+export const ImageDropdown = ({ imageValue, onChange, onOutsideClick }) => {
+	const fileInputRef = useRef();
+	const imageRef = useRef();
+	const containerRef = useRef();
+	const [url, setURL] = useState('');
+	const [showError, setShowError] = useState(false);
+	const [isDropping, setIsDropping] = useState(false);
+	const urlValueTimer = useRef(null);
+
+	const handleDelete = () => {
+		const image = imageRef.current;
+		if (image) {
+			onChange('');
+			setURL('');
+			setShowError(false);
+		}
+	};
+
+	const readFile = (file) => {
+		const reader = new FileReader();
```

Comment:
@SavvyShah Maybe we can abort the file reader and unsubscribe the registered event listeners

Reply:
No. When the reader is garbage collected, it's event listeners are removed.